Recast/PlayerViewController.swift

```diff
@@ -8,59 +8,344 @@
 
 import UIKit
 import SnapKit
+import AVFoundation
+
+private var playerViewControllerKVOContext = 0
 
 class PlayerViewController: UIViewController {
 
     // MARK: - Variables
+    @objc let player = AVQueuePlayer()
+
+    var currentTime: Double {
+        get {
+            return CMTimeGetSeconds(player.currentTime())
+        }
+        set {
+            let newTime = CMTimeMakeWithSeconds(newValue, preferredTimescale: 1)
+            player.seek(to: newTime, toleranceBefore: CMTime.zero, toleranceAfter: CMTime.zero)
+        }
+    }
+
+    var duration: Double {
+        guard let currentItem = player.currentItem else { return 0.0 }
+        return CMTimeGetSeconds(currentItem.duration)
+    }
+
+    var rate: Float {
+        get {
+            return player.rate
+        }
+        set {
+            player.rate = newValue
+        }
+    }
+
+    var playerLayer: AVPlayerLayer? {
+        return playerView.playerLayer
+    }
+
+    /*
+     A formatter for individual date components used to provide an appropriate
+     value for the `startTimeLabel` and `durationLabel`.
+     */
+    let timeRemainingFormatter: DateComponentsFormatter = {
+        let formatter = DateComponentsFormatter()
+        formatter.zeroFormattingBehavior = .pad
+        formatter.allowedUnits = [.minute, .second]
+        return formatter
+    }()
+
+    /*
+     A token obtained from calling `player`'s `addPeriodicTimeObserverForInterval(_:queue:usingBlock:)`
+     method.
+     */
+    var timeObserverToken: Any?
+
+    var durationObserverToken: Any?
+    var rateObserverToken: Any?
+    var statusObserverToken: Any?
+    var currentObserverToken: Any?
```

Comment:
why are these `Any?` instead of `NSKeyValueObservation?` ?

Reply:
This is what Apple had in their demo as well. I'm not sure. Since the fact it's not referenced as `NSKeyValueObservation` it's shorter to keep it around as this.